extensions/interactions/TextInput/TextInput.js

```diff
@@ -81,3 +82,58 @@ oppia.directive('oppiaShortResponseTextInput', [
     };
   }
 ]);
+
+oppia.factory('textInputRulesService', ['$filter', function($filter) {
+  return {
+    Equals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      return normalizedAnswer.toLowerCase() == normalizedInput.toLowerCase();
+    },
+    FuzzyEquals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var answerString = normalizedAnswer.toLowerCase();
+
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      var inputString = normalizedInput.toLowerCase();
+
+      if(inputString == answerString) return true;
```

Comment:
Also, please instead do:

      if (inputString == answerString) {
        return true;
      }

All bodies of if/else statements should be encased in braces.